test/qa/lib/ssh.py

```diff
@@ -37,6 +43,60 @@ def download_file(self, remote_path: str, path: str):
     def disconnect(self):
         self.client.close()
 
+    class Network:
+        def __init__(self, ssh_class_instance):
+            self.ssh_class_instance: Ssh = ssh_class_instance
+
+        def _is_internet_reachable(self, retry=5) -> bool:
+            """ returns True when remote host is reachable by it's public IP. """
+            i = 0
+            while i < retry:
```

Comment:
Maybe `poll` function(`nordvpn-linux/test/qa/lib/__init__.py`) could be used here? Example of usage: https://github.com/NordSecurity/nordvpn-linux/blob/main/test/qa/test_fileshare.py#L183